src/Lexer.php

```diff
@@ -382,6 +382,62 @@ private function lexExpression(): void
         }
     }
 
+    private function stripcslashes(string $str, string $quoteType): string {
+        $result = '';
+        $length = strlen($str);
+        $specialChars = [
+            'f' => "\f",
+            'n' => "\n",
+            'r' => "\r",
+            't' => "\t",
+            'v' => "\v",
+        ];
+
+        for ($i = 0; $i < $length; $i++) {
+            if ($str[$i] === '\\' && $i + 1 < $length) {
+                $nextChar = $str[$i + 1];
+                if (isset($specialChars[$nextChar])) {
+                    $result .= $specialChars[$nextChar];
+                    $i++;
+                } elseif ($nextChar === '\\') {
+                    $result .= '\\';
+                    $i++;
+                } elseif ($nextChar === "'" || $nextChar === '"') {
+                    if ($nextChar !== $quoteType) {
+                        trigger_deprecation('twig/twig', '3.11', 'Character "%s" at position %d does not need to be escaped anymore.', $nextChar, $i + 2);
+                    }
+                    $result .= $nextChar;
+                    $i++;
+                } elseif ($nextChar === '#' && $i + 2 < $length && $str[$i + 2] === '{') {
+                    $result .= '#{';
+                    $i += 2;
```

Comment:
I added this case, to make sure that escaping string Interpolation works. 

Should we also document this?

Reply:
> this code is needed to avoid the deprecation in case a string interpolation is escaped, right

Exactly.